lib/sierra-models/base.js

```diff
@@ -147,42 +142,47 @@ class SierraBase {
 
     if (Array.isArray(rawFields) && rawFields.length) {
       const vals = rawFields.filter(opts.preFilter).map((field) => {
-        if (field.parallel) {
-          const subfield6 = this._parseSubfield6(field.parallel)
-          // coerce parallel into array for the purpose of recursive call
-          field.parallel = this._prettifyRawVarFields([field.parallel], subfields, opts)[0]
-          if (subfield6) {
-            field.parallel.script = subfield6.script
-            field.parallel.direction = subfield6.direction
+        // Build return:
+        const varFieldMatchReturn = {}
+
+        // Build value
+        //
+        // If this raw varfield contains any subfields at all (always true, except for orphaned parallels):
+        // (sometimes there's a case error)
+        const unfilteredSubFields = field.subFields || field.subfields
+        if (unfilteredSubFields) {
+          let filteredSubfields = unfilteredSubFields
+
+          // Apply inclusion/exclusion rules to select desired subfields:
+          if (whichSubfields) {
+            filteredSubfields = filteredSubfields
+              .filter((sub) => whichSubfields.indexOf(sub.tag) >= 0)
+          } else if (opts.excludedSubfields) {
+            filteredSubfields = filteredSubfields
+              .filter((sub) => !opts.excludedSubfields.includes(sub.tag) && sub.tag !== '6')
           }
-        }
-        // sometimes there's a case error...
-        const unfilteredFields = field.subFields || field.subfields
-        // Remove any subfield we're meant to exclude, including subfield 6
-        const _subFields = unfilteredFields && opts.excludedSubfields
-          ? unfilteredFields.filter(subfield => {
-            return !opts.excludedSubfields.includes(subfield.tag) && subfield.tag !== '6'
-          })
-          : unfilteredFields
-
-        const subfieldParse = (subs) => {
-          const subfieldMap = subs.reduce((hash, sub) => {
+
+          varFieldMatchReturn.value = filteredSubfields.map((sub) => sub.content).join(opts.subfieldJoiner)
+          varFieldMatchReturn.subfieldMap = filteredSubfields.reduce((hash, sub) => {
             hash[sub.tag] = sub.content
             return hash
           }, {})
-          const value = subs.map((sub) => sub.content).join(opts.subfieldJoiner)
-          if (field.parallel) return { value, subfieldMap, parallel: field.parallel }
-          else return { value, subfieldMap }
+        } else if (field.content) {
+          varFieldMatchReturn.value = field.content
         }
 
-        // If asked to match certain subfields, return only those:
-        if (subfields) {
-          const subs = (_subFields || []).filter((sub) => subfields.indexOf(sub.tag) >= 0)
-          return subfieldParse(subs)
-          // Otherwise, attempt to return 'content', falling back on subfields' content:
-        } else {
-          return field.content || (_subFields ? subfieldParse(_subFields) : null)
```

Comment:
This legacy logic has grown confusing. The `if(subfields)` check is really a "if querying by specific subfields". Proposing we:
 - isolate (and centralize) the logic around selecting specific subfields either by inclusion or exclusion
 - remove the inline "subfieldParse" function (which has evolved to operate as a "render as varfieldmatch object" function), because it shouldn't be necessary after above
 - generally flatten the structure to clarify how we build the primary value, the subfieldMap, and the parallel prop

Reply:
sounds good to me!